Sources/FirebaseFirestore/DocumentReference+Swift.swift

```diff
@@ -48,7 +48,7 @@ extension DocumentReference {
       future.Wait(firebase.FutureBase.kWaitTimeoutInfinite)
     }
 
-    return snapshot.pointee.is_valid() ? snapshot.pointee : nil
+    return snapshot.pointee.exists && snapshot.pointee.is_valid() ? snapshot.pointee : nil
```

Comment:
How about guarding that the snapshot exists first?

```suggestion
    guard snapshot.pointee.exists else { return nil }
    return snapshot.pointee.is_valid() ? snapshot.pointee : nil
```